### PR TITLE
new(ci): build latest mainline kernel (even RC!) in latest-kernel job.

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area CI

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

Improves `latest-kernel` CI to properly test latest mainline kernel, RC included, so that we catch drivers build issues as soon as possible.
For example, i just caught a new issue by https://github.com/torvalds/linux/commit/dcfbb67e48a2becfce7990386e985b9c45098ee5 commit:
```
DEBU + make CC=/usr/bin/gcc-12.0.0 KERNELDIR=/tmp/kernel
DEBU *make -C /tmp/kernel M=/tmp/driver modules
DEBU *make[1]: Entering directory '/tmp/kernel'
DEBU �warning: the compiler differs from the one used to build the kernel
DEBU   The kernel was built by: gcc (Debian 12.2.0-14) 12.2.0
DEBU   You are using:           gcc-12.0.0 (Debian 12.2.0-14) 12.2.0
DEBU   CC [M]  /tmp/driver/main.o
DEBU �In file included from ./include/linux/linkage.h:7,
DEBU                  from ./include/linux/kernel.h:17,
DEBU                  from ./include/linux/cpumask.h:10,
DEBU                  from ./include/linux/smp.h:13,
DEBU                  from ./include/linux/tracepoint.h:15,
DEBU                  from ./include/trace/syscall.h:5,
DEBU                  from /tmp/driver/main.c:13:
DEBU /tmp/driver/main.c: In function 'scap_init':
DEBU ./include/linux/export.h:27:22: error: passing argument 1 of 'class_create' from incompatible pointer type [-Werror=incompatible-pointer-types]
DEBU    27 | #define THIS_MODULE (&__this_module)
DEBU       |                     ~^~~~~~~~~~~~~~~
DEBU       |                      |
DEBU       |                      struct module *
DEBU /tmp/driver/main.c:2915:36: note: in expansion of macro 'THIS_MODULE'
DEBU  2915 |         g_ppm_class = class_create(THIS_MODULE, DRIVER_DEVICE_NAME);
DEBU       |                                    ^~~~~~~~~~~
DEBU �In file included from ./include/linux/device.h:31,
DEBU                  from ./include/linux/node.h:18,
DEBU                  from ./include/linux/cpu.h:17,
DEBU                  from ./include/linux/static_call.h:135,
DEBU                  from ./include/linux/tracepoint.h:22:
DEBU ./include/linux/device/class.h:229:54: note: expected 'const char *' but argument is of type 'struct module *'
DEBU   229 | struct class * __must_check class_create(const char *name);
DEBU       |                                          ~~~~~~~~~~~~^~~~
DEBU y/tmp/driver/main.c:2915:23: error: too many arguments to function 'class_create'
DEBU  2915 |         g_ppm_class = class_create(THIS_MODULE, DRIVER_DEVICE_NAME);
DEBU       |                       ^~~~~~~~~~~~
DEBU ./include/linux/device/class.h:229:29: note: declared here
DEBU   229 | struct class * __must_check class_create(const char *name);
DEBU       |                             ^~~~~~~~~~~~
DEBU +cc1: some warnings being treated as errors
DEBU Fmake[2]: *** [scripts/Makefile.build:252: /tmp/driver/main.o] Error 1
DEBU 2make[1]: *** [Makefile:2026: /tmp/driver] Error 2
DEBU )make[1]: Leaving directory '/tmp/kernel'
DEBU $make: *** [Makefile:7: all] Error 2
DEBU log pipe close                                error=EOF
DEBU context canceled
```

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

NOTE: we need a new driverkit release once https://github.com/falcosecurity/driverkit/pull/272 gets merged, since driverkit did not support RC releases in vanilla target.
I'll keep this wip until we have the new release out.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/latest-kernel.yml [hunk 1]
```diff
@@ -20,24 +20,34 @@ jobs:
...
-            line=$(grep "pkgver" PKGBUILD | head -n1)
-            krel=${line#*=}
-            line=$(grep "pkgrel" PKGBUILD | head -n1)
-            kver=${line#*=}
-            echo "latest_vers=${krel}-${kver}" >> $GITHUB_OUTPUT
+            cd linux/
+            krel=$(grep "_tag" PKGBUILD | head -n1 | awk -F"=" '{print $2}')
+            echo "latest_vers=${krel:1}" >> $GITHUB_OUTPUT
+            echo "use_vanilla=true" >> $GITHUB_OUTPUT
+            echo "kernelconfig=$(cat config | base64 | tr -d '\n')" >> $GITHUB_OUTPUT
```
Comment (FedeDP) [inline .github/workflows/latest-kernel.yml:37]: I am not sure how `GITHUB_OUTPUT` will behave pushing lots of megabytes string into it. :pray:

**Comment by Andreagit97** [comment]
this job is ok but we need to keep in consideration that the content of an RC could change so we should probably wait for the actual tag to fix drivers or at least evaluate if something could still change in the next RC

**Comment by FedeDP** [comment]
Note however that RC should already be fix-only releases given https://www.kernel.org/category/releases.html, quote:
> after the merge window closes, there is a 7-week bugfix and stabilization period with weekly "release candidate" snapshots

Therefore I don't expect API breaks in between RCs :)

**Comment by FedeDP** [comment]
I was able to test the workflow (that is failing): https://github.com/falcosecurity/libs/actions/runs/4978292512/jobs/8908401147?pr=1090

**Comment by FedeDP** [comment]
Also, we might want to add support to generate dynamic badge for README (like "Test drivers build against 6.4-rc2 mainline kernel") using https://github.com/Schneegans/dynamic-badges-action.
This is some future work IMHO, it would be cool and does not require write access to the repo.

**Comment by FedeDP** [comment]
Since we now have a new driverkit release (https://github.com/falcosecurity/driverkit/releases/tag/v0.13.0), we can go on with this.
It has no impact on the release workflow, therefore we could even merge it now :)
/cc @jasondellaluce @Andreagit97

**Comment by Andreagit97** [comment]
I would still wait few days if for you it's ok, let's try to keep master and the release branch as synced as possible

**Comment by FedeDP** [comment]
No problem, i obey the release manager! :joy:

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by jasondellaluce — APPROVED** [review]
/approve

**Comment by FedeDP** [comment]
/unhold
Let's go with this one!

**Comment by FedeDP** [comment]
Closing and reopening :) 
/close

**Pull request merged**